document/core/exec/instructions.rst

```diff
@@ -690,6 +691,318 @@ Memory Instructions
    \end{array}
 
 
+.. _exec-memory.drop:
+
+:math:`\MEMORYDROP~x`
+.....................
+
+1. Let :math:`F` be the :ref:`current <exec-notation-textual>` :ref:`frame <syntax-frame>`.
+
+2. Assert: due to :ref:`validation <valid-memory.drop>`, :math:`F.\AMODULE.\MIMEMS[0]` exists.
+
+3. Let :math:`ma` be the :ref:`memory address <syntax-memaddr>` :math:`F.\AMODULE.\MIMEMS[0]`.
+
+4. Assert: due to :ref:`validation <valid-memory.drop>`, :math:`S.\SMEMS[ma]` exists.
+
+5. Let :math:`\X{mem}` be the :ref:`memory instance <syntax-meminst>` :math:`S.\SMEMS[ma]`.
+
+6. Assert: due to :ref:`validation <valid-memory.drop>`, :math:`F.\AMODULE.\MIDATAS[x]` exists.
+
+7. Let :math:`da` be the :ref:`data segment address <syntax-dataaddr>` :math:`F.\AMODULE.\MIDATAS[x]`.
+
+8. Replace :math:`S.\SDATA[da]` with :math:`\epsilon`.
+
+.. math::
+   ~\\[-1ex]
+   \begin{array}{l}
+   \begin{array}{lcl@{\qquad}l}
+   S; F; (\MEMORYDROP~x) &\stepto& S'; F; \epsilon
+   \end{array}
+   \\ \qquad
+     \begin{array}[j]{@{}r@{~}l@{}}
+     (\iff & F.\AMODULE.\MIMEMS[0] = ma \\
+     \wedge & F.\AMODULE.\MIDATAS[x] = da \\
+     \wedge & S' = S \with \SDATA[da] = \epsilon) \\
```

Comment:
Setting a SDATA slot to epsilon must allow for it in the definition of store, i.e., change the syntax to (\datainst^?)^*  (same for element segments).